File: Spring Boot/car_rental_service/src/main/java/com/car_rental_service/services/admin/AdminService.java
package com.car_rental_service.services.admin;

import java.io.IOException;
import java.util.List;

import com.car_rental_service.dto.BookACarDto;
import com.car_rental_service.dto.CarDto;
import com.car_rental_service.dto.CarDtoListDto;
import com.car_rental_service.dto.SearchCarDto;

public interface AdminService {

	boolean postCar(CarDto carDto) throws IOException;
	
	List<CarDto> getAllCars();
	
	void deleteCar(Long id);
	
	CarDto getCarById(Long id);
	
	boolean updateCar(Long carId, CarDto carDto) throws IOException;
	
	List<BookACarDto> getBookings();
	
	boolean changeBookingStatus(Long bookingId, String status);
	
	CarDtoListDto searchCar(SearchCarDto searchCarDto);
}

File: Angular/car-rental-service/src/app/modules/admin/service/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from 'src/app/auth/service/storage/storage.service';

const BASE_URL = ["http://localhost:8080"];

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(
    private http: HttpClient,
  ) { }

  postCar(carDto: any): Observable<any> {
    return this.http.post(`${BASE_URL}/api/admin/car`, carDto, {
      headers: this.createAuthorization()
    });
  }

  getAllCars(): Observable<any> {
    return this.http.get(`${BASE_URL}/api/admin/cars`, {
      headers: this.createAuthorization()
    });
  }

  deleteCar(id: number): Observable<any> {
    return this.http.delete(`${BASE_URL}/api/admin/car/${id}`, {
      headers: this.createAuthorization()
    });
  }

  getCarById(id: number): Observable<any> {
    return this.http.get(`${BASE_URL}/api/admin/car/${id}`, {
      headers: this.createAuthorization()
    });
  }

  updateCar(carId: number,carDto: any): Observable<any> {
    return this.http.put(`${BASE_URL}/api/admin/car/${carId}`, carDto, {
      headers: this.createAuthorization()
    });
  }

  getCarBookings(): Observable<any> {
    return this.http.get(`${BASE_URL}/api/admin/car/bookings`, {
      headers: this.createAuthorization()
    });
  }

  changeBookingStatus(bookingId: number, status:string): Observable<any> {
    return this.http.get(`${BASE_URL}/api/admin/car/booking/${bookingId}/${status}`, {
      headers: this.createAuthorization()
    });
  }

  searchCar(SearchCarDto: any): Observable<any> {
    return this.http.post(`${BASE_URL}/api/admin/car/search`, SearchCarDto, {
      headers: this.createAuthorization()
    });
  }

  createAuthorization(): HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + StorageService.getToken()
    );
  }
}
